packages/react-native/Libraries/Animated/useAnimatedProps.js

```diff
@@ -137,6 +137,14 @@ function reduceAnimatedProps<TProps>(
 function useAnimatedPropsLifecycle(node: AnimatedProps): void {
   const prevNodeRef = useRef<?AnimatedProps>(null);
   const isUnmountingRef = useRef<boolean>(false);
+  const hasMountedRef = useRef<boolean>(false);
+  const hasInitiallyAttached = useRef<boolean>(false);
+
+  // Attach animated node instantly, avoiding issues with native animations
+  if (!hasMountedRef.current) {
+    hasMountedRef.current = true;
+    node.__attach();
```

Comment:
The reason that `node.__attach();` was being executed in `useLayoutEffect()` is due to potential bugs with calling this outside of an effect when concurrent rendering is in effect.

With concurrent rendering, a component (and any hooks) may be evaluated multiple times before the views are actually created and effects executed. This is why `useEffect` and `useLayoutEffect` are called "effects"; they are side effects of the component.

But I don't yet have a good suggestion for how to do this "the right way".

Let me think about this a bit and get back to you…

Reply:
Thanks for the response and kind words @yungsters - appreciate you looking into it!

Just to quickly reiterate some of my comments I made in the linked issue for anyone else taking a look - the idea behind my fix came from trying to recreate the previous UNSAFE_componentWillMount() behaviour in <0.72’s [createAnimatedComponent.js](https://github.com/facebook/react-native/blob/0.71-stable/Libraries/Animated/createAnimatedComponent.js) which attached props immediately and seemed to help avoid the problem:

```js
UNSAFE_componentWillMount() {
  this._waitForUpdate();
  this._attachProps(this.props);
}
```

Anyway - totally understand if my update isn’t the best way to tackle the problem though. I hadn’t considered any issues with concurrent rendering, but I had the feeling going outside hooks wasn’t going to be desirable… Just couldn’t figure out a way around it besides using the ref's.

Not sure what else I can offer but please don’t hesitate to reach out if I can be of any assistance.